pkg/backend/identityapi/register_client.go

```diff
@@ -42,34 +61,121 @@ type RegisterEntity struct {
 }
 
 type RegisterEntityResponse struct {
-	ID  entity.ID  `json:"entityID"`
-	Key entity.Key `json:"entityKey"`
+	ID   entity.ID  `json:"entityID"`
+	Key  entity.Key `json:"entityKey"`
+	Name string     `json:"entityName"`
 }
 
+type RegisterBatchEntityResponse []RegisterEntityResponse
+
 func NewRegisterEntity(key entity.Key) RegisterEntity {
 	return RegisterEntity{key, "", "", nil, nil}
 }
 
-func NewIdentityRegisterClient(
+// NewRegisterClient returns an implementation of RegisterClient
+func NewRegisterClient(
 	svcUrl, licenseKey, userAgent string,
 	compressionLevel int,
 	httpClient backendhttp.Client,
-) (IdentityRegisterClient, error) {
+) (RegisterClient, error) {
 	if compressionLevel < gzip.NoCompression || compressionLevel > gzip.BestCompression {
 		return nil, fmt.Errorf("gzip: invalid compression level: %d", compressionLevel)
 	}
+	icfg := identity.NewConfiguration()
+	icfg.BasePath = svcUrl + identityPath
+	icfg.Debug = true
+	// TODO: add the global HTTP client here
+	// icfg.HTTPClient = httpClient
+	identityClient := identity.NewAPIClient(icfg)
 	return &registerClient{
 		svcUrl:           strings.TrimSuffix(svcUrl, "/"),
 		licenseKey:       licenseKey,
 		userAgent:        userAgent,
 		httpClient:       httpClient,
 		compressionLevel: compressionLevel,
+		apiClient:        identityClient.DefaultApi,
 	}, nil
 }
 
+func (rc *registerClient) RegisterBatchEntities(agentEntityID entity.ID, entities []protocol.Entity) (resp RegisterBatchEntityResponse, duration time.Duration, err error) {
+
+	ctx := context.Background()
```

Comment:
ctx should be an arg, so it can be cancelled

Reply:
Yep, was looking at that as at the moment where we call RegisterBatchEntities we don't have a ctx. This is something I'd like to improve but might me more work for this PR.

Reply:
added ctx cancellation as nice to have into feature branch https://github.com/newrelic/infrastructure-agent/pull/53